### PR TITLE
fix visual issues on dashboard tabs

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Part of epic https://github.com/metabase/metabase/issues/29502

### Description

This PR fixes a few small visual issues (font sizes, alignments, etc) for dashboard tabs. These are outlined in [this document](https://www.notion.so/metabase/Bring-tabbed-navigation-to-dashboards-dd2e71b4d1954f8188ecbd9914e81420#17219475a71c40529cf04ff41d1d4080) and [this thread](https://metaboat.slack.com/archives/C057T1QTB3L/p1684936928237019).

### How to verify

1. Create dashboard with tabs
2. Verify styles match [figma](https://www.figma.com/file/DuwiUCVo1K4EUBw50pGYPy/Tabs-in-dashboards?type=design&node-id=104-5378&t=MuhT58CrxPQCanOn-0)

### Demo

https://github.com/metabase/metabase/assets/37751258/e9cd39a1-87fc-4fee-bb22-a5f531a16f24

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

_Can't really test style changes and there are existing unit and e2e tests for this feature._

### CODE REVIEW

File: frontend/src/metabase/dashboard/actions/tabs.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ export const tabsReducer = createReducer<DashboardState>(
           const newTab = getDefaultTab({
             tabId,
             dashId,
-            name: t`Page ${prevTabs.length + 1}`,
+            name: t`Page ${prevTabs.filter(t => !t.isRemoved).length + 1}`,
```
Comment (EmmadUsmani) [inline frontend/src/metabase/dashboard/actions/tabs.ts:131]: There was a bug where new tabs would have the wrong default name if you previously deleted some tabs in the same edit mode session

**Review by kdoh — APPROVED** [review]
Looking way better in FF now. 👍

File: frontend/src/metabase/core/components/TabButton/TabButton.tsx [hunk 2]
```diff
@@ -179,6 +184,24 @@ export interface RenameableTabButtonProps<T>
   canRename?: boolean;
 }
 
+// These styles need to be here instead of .styled to avoid circular dependency
+const borderStyle = css`
+  border: 1px solid ${color("brand")};
+  box-shadow: 0px 0px 0px 1px ${lighten(color("brand"), 0.28)};
+`;
+export const RenameableTabButtonStyled = styled(_TabButton)<{
```
Comment (alxnddr) [inline frontend/src/metabase/core/components/TabButton/TabButton.tsx:192]: We try to avoid that and create a separate file for `RenameableTabButton`

**Review by alxnddr — APPROVED** [review]
Looks good, one optional comment

**Pull request merged**